platform/darwin/src/MGLGeometry_Private.h

```diff
@@ -13,6 +13,18 @@
 CGRect MGLExtendRect(CGRect rect, CGPoint point);
 
 NS_INLINE mbgl::LatLng MGLLatLngFromLocationCoordinate2D(CLLocationCoordinate2D coordinate) {
+    if (std::isnan(coordinate.latitude)) {
+        [NSException raise:NSInvalidArgumentException format:@"latitude must not be NaN"];
+    }
+    if (std::isnan(coordinate.longitude)) {
+        [NSException raise:NSInvalidArgumentException format:@"longitude must not be NaN"];
+    }
+    if (std::abs(coordinate.latitude) > 90.0) {
```

Comment:
ea20dd58e66981e39da87e3dfea663494d746011 adds preflight checks to some of the call sites above that avoid the exception and instead return special values indicating invalidity. If you try to convert an invalid coordinate to a point, you get an invalid point. You can create a camera with an invalid coordinate, but you’ll get an exception when you try to move the map to that camera. You can create a polyline, polygon, or point collection with an invalid coordinate, and the overlay will be invalid, but you’ll get an exception when you try to add that shape to the map.